controllers/cluster.weave.works/existinginframachine_controller.go

```diff
@@ -1024,8 +1024,22 @@ func (r *ExistingInfraMachineReconciler) SetupWithManager(mgr ctrl.Manager) erro
 // MachineControllerParams groups required inputs to create a machine actuator.
 type MachineControllerParams struct {
 	Client              client.Client
+	Log                 logr.Logger
+	Scheme              *runtime.Scheme
 	ClientSet           *kubernetes.Clientset
 	ControllerNamespace string
 	EventRecorder       record.EventRecorder
 	Verbose             bool
 }
+
+func NewMachineController(params *MachineControllerParams) *ExistingInfraMachineReconciler {
+	return &ExistingInfraMachineReconciler{
+		Client:              params.Client,
+		Log:                 params.Log,
```

Comment:
great quick hack, but shall we consider making clientSet/controllerNamespace/eventRecorder/verbose all publicly-settable with a `Legacy` prefix in order to not duplicate the params here and have more code than needed?

Reply:
This is actually not my quick hack, this is how it was implemented in wksctl upstream. I agree, this should not be a permanent solution, but I don't want to start renaming fields in `ExistingInfraMachineReconciler` just yet so that the code can be recognized. So for now it can stay IMO.